### PR TITLE
Use slot to determine fork version

### PR DESCRIPTION
**Pull request by saolyn**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

This PR updates the logic to use the slot instead of the head state version to determine the fork version.

This approach reduces the risk of errors in cases where the block for the first slot of a new fork is delayed or not proposed, which could result in the head state still reflecting the previous fork.

**Which issues(s) does this PR fix?**
N/A

**Other notes for review**

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have made an appropriate entry to [CHANGELOG.md](https://github.com/prysmaticlabs/prysm/blob/develop/CHANGELOG.md).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

### CODE REVIEW

**Review by james-prysm — DISMISSED** [review]

File: beacon-chain/rpc/eth/beacon/handlers_pool.go [hunk 1]
```diff
@@ -726,18 +720,19 @@ func (s *Server) GetAttesterSlashingsV2(w http.ResponseWriter, r *http.Request)
 	ctx, span := trace.StartSpan(r.Context(), "beacon.GetAttesterSlashingsV2")
 	defer span.End()
 
+	slot := s.HeadFetcher.HeadSlot()
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/handlers_pool.go:723]: Head slot is the slot of the most recent block, not the current slot. There is a `TimeFetcher` interface that can give you the current slot.

**Review by rkapka — CHANGES_REQUESTED** [review]

File: time/slots/slottime.go [hunk 2]
```diff
@@ -81,6 +82,25 @@ func ToEpoch(slot primitives.Slot) primitives.Epoch {
...
+		return version.Capella
+	case epoch >= params.BeaconConfig().BellatrixForkEpoch:
+		return version.Bellatrix
+	case epoch >= params.BeaconConfig().AltairForkEpoch:
+		return version.Altair
+	default:
+		return version.Phase0
+	}
+}
+
```
Comment (prestonvanloon) [inline time/slots/slottime.go:103]: This is a good candidate for a unit test.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Looks like a big improvement to avoid reading from the state. I am a bit concerned that there are no unit tests added or changed. Could you add some tests for this logic change?

File: time/slots/slottime.go (hunk 2 above)
Comment (saolyn) [reply time/slots/slottime.go:103]: Added

File: beacon-chain/rpc/eth/beacon/handlers_pool.go [hunk 3]
```diff
@@ -98,13 +98,10 @@ func (s *Server) ListAttestationsV2(w http.ResponseWriter, r *http.Request) {
 	if !ok {
 		return
 	}
-
-	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
-	if err != nil {
-		httputil.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
-		return
+	v := slots.ToForkVersion(primitives.Slot(slot))
+	if slot == 0 {
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/beacon/handlers_pool.go:102]: shouldn't this be based on rawSlot instead of slot?

**Review by rkapka — APPROVED** [review]
LGTM

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**